lldb/tools/lldb-dap/SourceBreakpoint.cpp

```diff
@@ -12,15 +12,19 @@
 
 namespace lldb_dap {
 
-SourceBreakpoint::SourceBreakpoint(const llvm::json::Object &obj)
-    : Breakpoint(obj), logMessage(std::string(GetString(obj, "logMessage"))),
+SourceBreakpoint::SourceBreakpoint(DAP *dap, const llvm::json::Object &obj)
+    : Breakpoint(dap, obj),
+      logMessage(std::string(GetString(obj, "logMessage"))),
       line(GetUnsigned(obj, "line", 0)), column(GetUnsigned(obj, "column", 0)) {
 }
 
 void SourceBreakpoint::SetBreakpoint(const llvm::StringRef source_path) {
+  if (!dap)
```

Comment:
when would that happen?

Reply:
It would happen if a breakpoint outlived the dap or was incorrectly initialized. However, I replaced the `DAP*` with a `DAP&`, which means this shouldn't happen anymore.